libs/mutantschemata/cpp_source/api.cpp

```diff
@@ -14,51 +14,14 @@ in order to call D code and insert/select mutants from db obtained from Dextool
 #include "rewrite.hpp"
 
 #include <iostream>
- #include <cstring>
+#include <cstring>
 
 void runSchemataCpp(SchemataApiCpp *sac, CppString::CppStr cs){
-/*
-    CppString::CppStr include = CppString::getStr("fibonacci_example.hpp"); // the file included with "#include "fibonacci_example.hpp""
-    std::cout << "trying findInclude from Cpp-side" << std::endl;
-    CppString::CppStr res = sac->apiFindInclude(cs, include);
-    std::cout << *res.cppStr << std::endl;
-
-    // temporary tests
-    {   // using CppBytes
-        sac->apiBuildMutant();
-        CppType::SchemataMutant before_edit = sac->apiSelectMutant(CppString::getBytes("id == 101", 9));
-
-        // change inject and insert
-        before_edit.inject = 5;
-        sac->apiInsertSchemataMutant(before_edit);
-
-        // select updated mutant
-        CppType::SchemataMutant after_edit = sac->apiSelectSchemataMutant(CppString::getBytes("\"loc.line\" = 14", 15));
-
-        after_edit.print();
-    }
-    {   // using CppStr
-        sac->apiBuildMutant();
-        CppType::SchemataMutant before_edit = sac->apiSelectMutant(CppString::getStr("id == 101"));
-
-        // change inject and insert
-        before_edit.inject = 97;
-        before_edit.loc.line = 900;
-        sac->apiInsertSchemataMutant(before_edit);
-
-        // select updated mutant
-        CppType::SchemataMutant after_edit = sac->apiSelectSchemataMutant(CppString::getStr("\"loc.line\" = 900"));
-
-        after_edit.print();
-    }
-*/
-  std::cout << "here is: " << *cs.cppStr << std::endl;
-        //llvm::errs() << "Usage: rewritersample <file,otherfile,...> includeDir workingDir\n";
-       // return 1;
-
-  char cstr[cs.cppStr->size()+1];
-  strcpy(cstr, cs.cppStr->c_str());
-
-  setupClang(cstr, ".", ".");
+    std::cout << "here is: " << *cs.cppStr << std::endl;
+    //llvm::errs() << "Usage: rewritersample <file,otherfile,...> includeDir workingDir\n";
+    // return 1;
+    char cstr[cs.cppStr->size()+1];
+    strcpy(cstr, cs.cppStr->c_str());
```

Comment:
Note that this is dangerous because cstr is stack allocated and a pointer to it escapes this function to setupClang. In this case it isn't a problem because setupClang copy the data to an internal stringstream.

`char cstr[cs.cppstr->size()+1]` will call [alloca](http://man7.org/linux/man-pages/man3/alloca.3.html) under the hood (see LLVM IR).